From 11d5e7052dfa46b74de9e526a613d7848115f169 Mon Sep 17 00:00:00 2001
From: "junwei.ma" <junwei.ma@amlogic.com>
Date: Thu, 3 Aug 2023 13:51:53 +0800
Subject: [PATCH 15/15] ISP: [isp][driver] SM1 1080P output RGB [1/1]

PD#SWPL-134178

Problem:
1080P can't output RGB

Solution:
1080P output RGB

Verify:
ok

Change-Id: I0d10a862a45d4980b8e57798dddcadc78b3b5abe
Signed-off-by: junwei.ma <junwei.ma@amlogic.com>
---
 drivers/media/camera/ov5640.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/drivers/media/camera/ov5640.c b/drivers/media/camera/ov5640.c
index f90f77054..84be52baf 100644
--- a/drivers/media/camera/ov5640.c
+++ b/drivers/media/camera/ov5640.c
@@ -1361,6 +1361,15 @@ static struct aml_camera_i2c_fig_s OV5640_capture_5M_script[] = {
 
 /*mipi lane bps : M*/
 static struct resolution_param  prev_resolution_array[] = {
+	{
+		.frmsize            = {2592, 1944},
+		.active_frmsize     = {2592, 1944},
+		.active_fps         = 15,
+		.lanes              = 2,
+		.bps                = 672,
+		.size_type          = SIZE_2592X1944,
+		.reg_script         = OV5640_capture_5M_script,
+	},
 	{
 		.frmsize            = {1920, 1080},
 		.active_frmsize     = {1920, 1080},
@@ -2648,7 +2657,7 @@ static int vidioc_s_fmt_vid_cap(struct file *file, void *priv,
 		fh->height, fh->width);
 
 	if (f->fmt.pix.pixelformat == V4L2_PIX_FMT_RGB24) {
-		res_param = get_resolution_param(dev, 1, fh->width, fh->height);
+		res_param = get_resolution_param(dev, 0, fh->width, fh->height);
 		if (!res_param) {
 			pr_err("error, resolution param not get\n");
 			goto out;
@@ -2851,14 +2860,16 @@ static int vidioc_enum_framesizes(struct file *file, void *fh,
 	if (fmt->fourcc == V4L2_PIX_FMT_NV21 ||
 			fmt->fourcc == V4L2_PIX_FMT_NV12 ||
 			fmt->fourcc == V4L2_PIX_FMT_YUV420 ||
-			fmt->fourcc == V4L2_PIX_FMT_YVU420) {
+			fmt->fourcc == V4L2_PIX_FMT_YVU420 ||
+			fmt->fourcc == V4L2_PIX_FMT_RGB24) {
 		if (fsize->index >= ARRAY_SIZE(prev_resolution_array))
 			return -EINVAL;
 		frmsize = &prev_resolution_array[fsize->index].frmsize;
 		fsize->type = V4L2_FRMSIZE_TYPE_DISCRETE;
 		fsize->discrete.width = frmsize->width;
 		fsize->discrete.height = frmsize->height;
-	} else if (fmt->fourcc == V4L2_PIX_FMT_RGB24) {
+	}
+		/* else if (fmt->fourcc == V4L2_PIX_FMT_RGB24) {
 		if (fsize->index >= ARRAY_SIZE(capture_resolution_array))
 			return -EINVAL;
 		frmsize =
@@ -2867,6 +2878,7 @@ static int vidioc_enum_framesizes(struct file *file, void *fh,
 		fsize->discrete.width = frmsize->width;
 		fsize->discrete.height = frmsize->height;
 	}
+*/
 	return ret;
 }
 
-- 
2.25.1

